src/pages/BooksList.jsx

```diff
@@ -0,0 +1,75 @@
+import axios from "axios";
+import { useState, useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+import "../styles/BooksList.css";
+
+function BooksList() {
+  const [booksData, setBooksData] = useState(null);
+  const [loading, setLoading] = useState(true);
+  const navigate = useNavigate();
+
+  const getListOfBooks = async () => {
```

Comment:
This function doesn't need any variable from the `BooksList` component scope. So you can define it outside the "body" of the component itself, to avoid redefining it on every render.